### PR TITLE
 RavenDB-11837 - Patching with counters fails to update the correct document

### PR DESCRIPTION
**Pull request by grisha-kotler**
http://issues.hibernatingrhinos.com/issue/RavenDB-11837

### CODE REVIEW

**Comment by grisha-kotler** [comment]
@aviv86 Can you review this?

**Review by aviv86 — COMMENTED** [review]
LGTM

File: src/Raven.Server/Documents/Patch/PatchDocumentCommand.cs [hunk 1]
```diff
@@ -165,9 +165,26 @@ public override int Execute(DocumentsOperationContext context)
                     return 1;
                 }
 
-                if (_run.UpdateDocumentCounters)
+                if (_run.UpdatedDocumentCounterIds != null)
                 {
-                    modifiedDocument = UpdateCountersInMetadata(context, modifiedDocument, originalDocument);
+                    foreach (var docId in _run.UpdatedDocumentCounterIds)
```
Comment (maximburyak) [inline src/Raven.Server/Documents/Patch/PatchDocumentCommand.cs:170]: Maybe we could sort those first

Comment (grisha-kotler) [reply src/Raven.Server/Documents/Patch/PatchDocumentCommand.cs:170]: Why does it matter?

**Pull request merged**